data_masks/user_redaction_config.go

```diff
@@ -0,0 +1,109 @@
+package data_masks
+
+import (
+	"regexp"
+	go_slices "slices"
+	"sync"
+
+	"github.com/akitasoftware/akita-libs/api_schema"
+	"github.com/akitasoftware/go-utils/sets"
+	"github.com/akitasoftware/go-utils/slices"
+	"github.com/postmanlabs/postman-insights-agent/printer"
+)
+
+type userRedactionConfig struct {
+	fieldNames       sets.Set[string]
+	fieldNameRegexps []*regexp.Regexp
+
+	// Protects this instance.
+	mu sync.RWMutex
+}
+
+// Creates a userRedactionConfig instance from the given ServiceAgentConfig.
+func newUserRedactionConfig(
+	agentConfig *api_schema.ServiceAgentConfig,
+) *userRedactionConfig {
+	return &userRedactionConfig{
+		fieldNames:       sets.NewSet(agentConfig.FieldsToRedact.FieldNames...),
+		fieldNameRegexps: agentConfig.FieldsToRedact.FieldNameRegexps,
+	}
+}
+
+// Determines whether fields with the given name should be redacted according to
+// this configuration.
+func (c *userRedactionConfig) redactsFieldsNamed(fieldName string) bool {
```

Comment:
Non-blocking: why `redacts` instead of `redact`?

Reply:
Okay, I see the usage becomes slightly non-idiomatic: `if opts.redactsFieldsNamed(...) { /* do redaction */ }`. Renamed in 08e6119.

Reply:
Yes, I read it as "does this config redact fields named X" or "do we redact the field named X"

I guess you were reading it more like "is it true that config redacts fields named X"?